lib/tests/test_conflicts.rs

```diff
@@ -944,6 +1100,26 @@ fn test_parse_conflict_malformed_diff() {
     );
 }
 
+#[test]
+fn test_parse_conflict_wrong_git_style() {
```

Comment:
Might also want one for this (i.e. "snapshot" format with the first `+++++++` line missing):

```
line 1
<<<<<<<
left
-------
base
+++++++
right
>>>>>>>
line 5
```

That seems like something that could easily be allowed when maybe it shouldn't. Might also want one test for a mix of "snapshot" and "git" delimiters. It seems unlikely that these cases would happen in practice but might be best to be strict to start? It's hard to make it more strict later if find a reason to want to make it more strict.

Reply:
I added tests for the cases you mentioned, and I made the parsing logic a bit more strict. Now it determines whether it's a Git-style conflict or a JJ-style conflict based on the first line of the hunk, and then parses it using the appropriate function. When parsing a JJ-style conflict, Git-style conflict markers are ignored; and when parsing a Git-style conflict, JJ-style conflict markers are ignored. Does this make sense to you?